catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/model/Association.js

```diff
@@ -142,15 +143,17 @@ define([
         },
         removeSegment: function (segment) {
             var model = this;
-            var updatedAssociations = [];
+            var removedAssociations = [];
             var associations = this.get('associations').models;
             var segId = segment.get('segmentId');
             _.each(associations, function (association) {
-                if (association.get('sourceId') !== segId && association.get('targetId') !== segId) {
-                    updatedAssociations.push(association);
+                if (association.get('sourceId') === segId || association.get('targetId') === segId) {
+                    removedAssociations.push(association);
                 }
             });
-            this.set('associations', updatedAssociations);
+            _.each(removedAssociations, function(association){
```

Comment:
I didn't want to remove elements from the array i was looping over. That is why I create an array of items to delete and the loop through those and delete them from the first array. I did clean it up using the built in find instead of the first loop.